File: src/programa.c
#include <stdio.h>

void entradaNombre(char *nombre, int longitud) {
    // Función para obtener el nombre de usuario
    printf("Introduce el nombre de usuario: ");
    fgets(nombre, longitud, stdin);
}

void salidaNombre(const char *nombre) {
    // Función para mostar el nombre de usuario
    printf("Hola %s.Este programa está hecho en el lenguaje de programación: C. \n", nombre);
}

int main() {
    char nombre[100];

    // Entrada
    entradaNombre(nombre, sizeof(nombre));
    // Salida
    salidaNombre(nombre);

    return 0;
}

File: src/programa.py
def entradaNombre():
    """Función que obtiene el nombre de usuario"""
    return input("Introduce el nombre de usuario: ")

def salidaNombre(nombre):
    """Función para mostar el nombre de usuario y lenguaje usado"""
    print(f"Hola {nombre}. Este programa está hecho en el lenguaje de programación: Python.")

def main():
   
    # Entrada
    nombre = entradaNombre()
    # Salida
    salidaNombre(nombre)

if __name__ == '__main__':
    main()